console/src/components/Services/Data/TableModify/ModifyTable.js

```diff
@@ -221,6 +221,69 @@ const ColumnEditor = ({
       </option>
     );
   }
+
+  const generateAlterOptions = datatypeOptions => {
+    return dataTypes.map((datatype, index) => {
+      console.log(datatype, datatypeOptions);
+      if (datatypeOptions.includes(datatype.value)) {
+        return (
+          <option
+            value={datatype.value}
+            key={index}
+            title={datatype.description}
+          >
+            {datatype.name}
+          </option>
+        );
+      }
+    });
+  };
+
+  const modifyAlterOptions = columntype => {
+    const integerOptions = [
+      'integer',
+      'serial',
+      'bigint',
+      'bigserial',
+      'numeric',
+      'text',
+    ];
+    const bigintOptions = ['bigint', 'bigserial', 'text', 'numeric'];
+    const uuidOptions = ['uuid', 'text'];
+    const jsonOptions = ['json', 'jsonb'];
+    switch (columntype) {
+      case 'integer':
+        return generateAlterOptions(integerOptions);
+
+      case 'serial':
+        return generateAlterOptions(integerOptions);
+
+      case 'bigint':
+        return generateAlterOptions(bigintOptions);
+
+      case 'bigserial':
+        return generateAlterOptions(bigintOptions);
+
+      case 'uuid':
+        return generateAlterOptions(uuidOptions);
+
+      case 'json':
+        return generateAlterOptions(jsonOptions);
+
+      case 'jsonb':
+        return generateAlterOptions(jsonOptions);
+
+      case 'timestamp with time zone':
```

Comment:
it will be great if you don't use plain strings
you could add columnTypes keys to [constants.js](https://github.com/hasura/graphql-engine/blob/master/console/src/constants.js) file and export them to use them in here like this example: 
```
// contants.js

export const INTEGER = 'integer';
export const SERIAL = 'serial';

// here
import { SERIAL, INTEGER } from 'constants';
switch (columntype) {
      case INTEGER:
        return generateAlterOptions(integerOptions);
       case SERIAL:
        return generateAlterOptions(integerOptions);

```